### PR TITLE
Add windows discord, multiline discord parser formats

### PR DESCRIPTION
**Pull request by Adjective-Object**
- generalize the multiline slack parser to accept a header regex and "extractor" that pulls an initial message from a regex match
- use that generalized parser to add parsers for multiline discord logs and the new (windows?) discord format

### CODE REVIEW

File: src/lib/classifyquote.tsx [hunk 1]
```diff
@@ -39,7 +70,7 @@ melanie
 [5:23 PM]
 I just got home, so I'm gonna read them now!
 */
-const slackHeaderRegex = /^(.+?[^\s])( |$\n?)\[\d{1,2}:\d{2} (A|P)M\](\s+)?/;
+const slackVerboseHeaderRegex = /^(.+?[^\s])( |$(\r\n|\n)?)\[\d{1,2}:\d{2} (A|P)M\](\s+)?/;
```
Comment (Speculative) [inline src/lib/classifyquote.tsx]: This is failing for me in this case:
```
melanie
[5:16 PM]
Qq
```

But it works if it looks like:
```
Tiffany [5:15 PM]
shucks

melanie
[5:16 PM]
Qq
```

Comment (Adjective-Object) [inline src/lib/classifyquote.tsx]: ah,
`( |$(\r\n|\n)?)` should be ` $(\r\n|\n)?`, I think?

**Comment by Adjective-Object** [comment]
should also turn the non-relevant capture groups into non-capturing groups as well

File: src/lib/classifyquote.tsx (hunk 1 above)
Comment (Speculative) [inline src/lib/classifyquote.tsx]: What's the `$` mean there? It's not EOL right?

Comment (Speculative) [inline src/lib/classifyquote.tsx]: Also I encountered an issue when testing the fancy display frontend thing yesterday. Things like
```
Tiffany [5:15 PM]
shucks

[5:15 PM]
shucks
```

Were turning the second timestamp into part of the message

**Comment by Speculative** [comment]
Alright kids @Adjective-Object @nolanlum we're in business as far as pastes and serialization are concerned. I've got it serializing to an irc-like format. No support for the new `props` field yet so we're just completely dropping the paste classification from the serialized quote. That can be added back in later via `props`.

Serialization format looks something like
```
<jeff> I said a thing
<nolm> I also said a thing
```

Preview: http://speculative.s-ul.eu/zLNkImjV

**Comment by nolanlum** [comment]
merging this because it's 3am and why not

**Pull request merged**